modules/hostap/src/supp_main.c

```diff
@@ -523,6 +585,456 @@ static int register_supplicant_event_socket(struct supplicant_context *ctx)
+			   errors, fname);
+		hostapd_config_free(conf);
+		conf = NULL;
+	}
+#endif /* WPA_IGNORE_CONFIG_ERRORS */
+	return conf;
+}
+
+static struct hostapd_iface *hostapd_interface_init(struct hapd_interfaces *interfaces,
+						    const char *if_name,
+						    const char *config_fname,
+						    int debug)
+{
+	struct hostapd_iface *iface;
+	int k;
+
+	wpa_printf(MSG_DEBUG, "Configuration file: %s", config_fname);
+	iface = hostapd_init(interfaces, config_fname);
+	if (!iface) {
+		return NULL;
+	}
+
+	if (if_name) {
+		os_strlcpy(iface->conf->bss[0]->iface, if_name,
+			   sizeof(iface->conf->bss[0]->iface));
+	}
+
+	iface->interfaces = interfaces;
+
+	for (k = 0; k < debug; k++) {
+		if (iface->bss[0]->conf->logger_stdout_level > 0) {
+			iface->bss[0]->conf->logger_stdout_level--;
+		}
+	}
+
+	if (iface->conf->bss[0]->iface[0] == '\0' &&
+	    !hostapd_drv_none(iface->bss[0])) {
+		wpa_printf(MSG_ERROR,
+			   "Interface name not specified in %s, nor by '-i' parameter",
+			   config_fname);
+		hostapd_interface_deinit_free(iface);
+		return NULL;
+	}
+
+	return iface;
+}
+
+static void zephyr_hostapd_init(struct supplicant_context *ctx)
+{
+	struct hapd_interfaces *interfaces = &ctx->hostapd;
+	size_t i;
+	int ret, debug = 0;
+	const struct device *dev;
+	struct net_if *iface;
+	char ifname[IFNAMSIZ + 1] = { 0 };
+	const char *entropy_file = NULL;
+	size_t num_bss_configs = 0;
+	int start_ifaces_in_sync = 0;
+#ifdef CONFIG_DPP
+	struct dpp_global_config dpp_conf;
+#endif /* CONFIG_DPP */
+
+	os_memset(interfaces, 0, sizeof(struct hapd_interfaces));
+	interfaces->reload_config      = hostapd_reload_config;
+	interfaces->config_read_cb     = hostapd_config_read2;
+	interfaces->for_each_interface = hostapd_for_each_interface;
+	interfaces->ctrl_iface_init    = NULL;
+	interfaces->ctrl_iface_deinit  = NULL;
+	interfaces->driver_init        = hostapd_driver_init;
+	interfaces->global_iface_path  = NULL;
+	interfaces->global_iface_name  = NULL;
+	interfaces->global_ctrl_sock   = -1;
+	dl_list_init(&interfaces->global_ctrl_dst);
+#ifdef CONFIG_DPP
+	os_memset(&dpp_conf, 0, sizeof(dpp_conf));
+	dpp_conf.cb_ctx = interfaces;
+	interfaces->dpp = dpp_global_init(&dpp_conf);
+	if (!interfaces->dpp) {
+		return;
+	}
+#endif /* CONFIG_DPP */
+
+	interfaces->count = 1;
+	if (interfaces->count || num_bss_configs) {
+		interfaces->iface = os_calloc(interfaces->count + num_bss_configs,
+					      sizeof(struct hostapd_iface *));
+		if (interfaces->iface == NULL) {
+			wpa_printf(MSG_ERROR, "malloc failed");
+			return;
+		}
+	}
+
+	if (hostapd_global_init(interfaces, entropy_file)) {
+		wpa_printf(MSG_ERROR, "Failed to initialize global context");
+		return;
+	}
+
+	eloop_register_timeout(HOSTAPD_CLEANUP_INTERVAL, 0,
+			       hostapd_periodic, interfaces, NULL);
+
+	iface = net_if_get_wifi_uap();
+	/* Allocate and parse configuration for full interface files */
+	dev = net_if_get_device(iface);
+	strncpy(ifname, dev->name, IFNAMSIZ);
```

Comment:
We should not use device name as an interface name.

Reply:
Updated.